numpyro/distributions/distribution.py

```diff
@@ -65,6 +102,12 @@ class Distribution(object):
     reparametrized_params = []
     _validate_args = False
 
+    @staticmethod
```

Comment:
Would just renaming this to `Distribution.set_args_validation(...)`, the default is kind of implicit at this level. Also, have we turned on distribution validation check? I think we should do that by default now.

Reply:
This is just to match PyTorch name, not important I guess?

> turn on validation

We can turn on by default but we should turn it off in SVI and MCMC because it will add overhead while give no effect. WDYT?

Reply:
This is not particularly important, since we don't use that `staticmethod` except from the utility methods in Pyro. I will leave it up to you in that case, I don't have a strong opinion. 

> We can turn on by default but we should turn it off in SVI and MCMC because it will add overhead while give no effect. WDYT?

Yeah, let us just keep it as is and advertise `numpyro.enable_validation` in our tutorials/examples instead.